core/storage/email/gae_models_test.py

```diff
@@ -128,9 +149,101 @@ def test_get_by_hash_behavior_with_sent_datetime_lower_bound(self):
                     sent_datetime_lower_bound='Not a datetime object')
 
 
+class ReplyToIdModelUnitTests(test_utils.GenericTestBase):
+    """Test the GeneralFeedbackEmailReplyToIdModel class."""
+
+    def setUp(self):
+        super(ReplyToIdModelUnitTests, self).setUp()
+
+        def _generate_unique_id_for_tests(unused_cls):
+            return 'Non_unique_id'
+
+        self.generate_constant_id_ctx = self.swap(
+            email_models.GeneralFeedbackEmailReplyToIdModel,
+            '_generate_unique_reply_to_id',
+            types.MethodType(
+                _generate_unique_id_for_tests,
+                email_models.GeneralFeedbackEmailReplyToIdModel))
+
+    def test_id_generation_works_correctly(self):
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created = model.create('user_id', 'thread_id')
+        self.assertEqual(created.id, 'user_id.thread_id')
+
+        created = model.create('other_user_id', 'other_thread_id')
+        self.assertNotEqual(created.id, 'user_id.thread_id')
+
+    def test_generated_reply_to_id_is_unique(self):
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        all_ids = [
+            model.create('user_id', 'thread_id').reply_to_id
+            for _ in xrange(100)]
+        set_ids = set(all_ids) #Set contains unique values
+        self.assertEqual(len(all_ids), len(set_ids))
+
+    def test_create_raises_when_duplicate_ids(self):
+        with self.generate_constant_id_ctx, self.assertRaises(Exception):
+            created1 = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+                'user_id', 'thread_id')
+            created1.put()
+            created2 = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+                'user_id', 'thread_id')
+            created2.put()
+
+    def test_get_by_reply_to_id_works_correctly(self):
```

Comment:
Can you be more specific in the name about what specific works correctly? Here & elsewhere: I recommend ensuring that each method only tests a single behavior, as that helps both with naming the test, and ensuring tests don't fail for unrelated changes to the corresponding production code. Beyond that, consider using the following format when naming tests:

test_action_withCondition_withAnotherCondition_hasExpectedOutcome

In this case, I see two potential tests that could be derived from this method:
- test_get_reply_to_id_can_be_used_to_retrieve_model
- test_nonexistent_get_reply_to_id_should_provide_no_model

Consider how you could also apply this one-behavior-per-method and test naming scheme elsewhere in this PR.

Reply:
Done, and done in other places I could think of